### PR TITLE
feat: add stats page

### LINKED ISSUE #10: Create Stats Page
This page should display statistics regarding the `IPFS` node.

Example:
- Size of the repo
- Fetched files

### PR DESCRIPTION
**Pull request by vasco-santos**
Created the MVP Stats Page for the IPFS Node running in the service worker.

For this, I inspired myself in the current version Web UI styling, as well as the dir-view

screen shot 2018-05-29 at 15 04 15

Closes #10

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
+  })
+}
+
+// Fetch stats
+const fetchStats = () => {
+  return new Promise((resolve, reject) => {
+    node.get()
+      .then((ipfsNode) => {
+        Promise.all([ipfsNode.id(), ipfsNode.repo.stat()])
+          .then((values) => {
```
Comment (hugomrdias) [inline src/index.js]:
```js
.then(([id,stat]) => {
```

File: src/index.js [hunk 2]
```diff
@@ -42,6 +84,13 @@ self.addEventListener('install', (event) => {
...
   node.get().then((ipfs) => {
     ipfsNode = ipfs
+
+    // Keep a record of the start date and time of the IPFS Node
+    const d = new Date()
+    startDateTime = {
+      date: `${d.getFullYear()}/${d.getMonth()}/${d.getDate()}`,
+      time: `${d.getHours()}:${d.getMinutes()}:${d.getSeconds()}`
+    }
   })
```
Comment (hugomrdias) [inline src/index.js]: add catch

File: src/index.js [hunk 3]
```diff
@@ -1,19 +1,54 @@
...
+let startDateTime = {
+  date: '',
+  time: ''
+}
 
-const fetchFile = (ipfsPath) => {
+// Fetch CID
+const fetchCID = (ipfsPath) => {
   return new Promise((resolve, reject) => {
     node.get()
```
Comment (hugomrdias) [inline src/index.js]: return here instead of creating a new Promise

File: src/index.js [hunk 4]
```diff
@@ -1,19 +1,54 @@
...
-      .then((resp) => resolve(resp))
+      .then((resp) => {
+        // Keep a record of the fetched CID (and fetch date)
+        const d = new Date()
+        fetchedCIDs.push({
+          cid: ipfsPath.split('/ipfs/')[1],
+          date: `${d.getFullYear()}/${d.getMonth()}/${d.getDate()}`,
+          time: `${d.getHours()}:${d.getMinutes()}:${d.getSeconds()}`
+        })
+        resolve(resp)
```
Comment (hugomrdias) [inline src/index.js]: and also return here instead of resolving

**Review by hugomrdias — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 5]
```diff
@@ -40,9 +74,18 @@ self.addEventListener('install', (event) => {
...
-    ipfsNode = ipfs
-  })
+  node.get()
+    .then((ipfs) => {
+      ipfsNode = ipfs
+
+      // Keep a record of the start date and time of the IPFS Node
+      const d = new Date()
+      startDateTime = {
+        date: `${d.getFullYear()}/${d.getMonth()}/${d.getDate()}`,
```
Comment (fsdiogo) [inline src/index.js]: As you're doing this twice, you could abstract it to a tiny method that returns the date and time. But up to you.

File: src/stats-view/style.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
...OX3VqoObVqy8E2nl3sAA0va0sOHID9PKMX5gsHZaRscPzGUBSD+/wfY2O56fcfo2jAQzrEs+wrp0bH5xZ3Q4cvqq733vMeee0lYEREAEREAEsoFAVgtA5wKdcM2vPplXUPDLQFVVzchwFKNAw85bU26Ly8vtsDGBysxMK5erAtD282tqtCK95kk48uHjl19UaA22tvZ379j+7Sd/+n9XTnmDaAcREAEREAERyEACOSEAeV3O37QpsO7u+74bqK78cmFJsJgZHjhCZFLoH8iRQI4IFgaDJlV8s0+uCcDo4KA94seRP1M/P073ckX48EB/rPPFTbcf9v4TPn3zokVh31xEdUQEREAEREAEEkwgZwSgw23x88/P2P3Aw78rq69/L4Qd0spBCBpOCzMESOmMOtgMOwiw06aft7kiAOnbN9DSAttj2aGADC4KhT0XeOD6j3Zu3PjMaHHBJ1Z++9tbDKpqFxEQAREQARHIaAI5JwCdq3XKPfe8rXvH7ttKamsPpHhwEzamqLQUo4H1WDVcjdBw+U6TvtxmuwBkTuhwB9K3NTdhBG/A+BrYfn5Y8NPXsLsl0tdz7opLL73XuLJ2FAEREAEREIEMJ5CzAtC5bidce/0ShPq4oriisnJkOGK8SpT1AxWVVmjWLIt+gn4t2SwA6d/X19hoDfV0G+O307cVIX1bR3u4Z9fOy1dd/sPLjCtrRxEQAREQARHIEgI5LwB5HZesbizd/MgdPw1UV51bFCgpcuUfCP+x0mnTMSJYh7RyJb67LbJRAMaGwhjxQ1iXtlZzP07Hz29wYKRry9Z7qmYe+qn7v3Rmj+8umDokAiIgAiIgAikgIAE4AfKSdevqtyx98M+ldfXvQGJY92nlkFu4dPp0CyuOJ7Sa3j+zSQCOIqj3QGurnbuXIt2owM+PvpuI5WN1bdm8rqAw72PLL7lkg1Fd7SQCIiACIiACWUpAAnCSC3vyrX86brCz46bgtGlzuaDAlX9gWZkdNqakqgqiI/14s0IAYpFOuKsLWTwareH+/kmu2OQv0c+P4q+voaED2T++9NgPvnvr5HvqVREQAREQARHILQL+GaryEfeNf7lz6+K33fvzHfmP9iCA9DHFZWVcKmq0Wphicaiz0xpG2rFCTAkzvEg6C0fKBtvaJu0CV8GyTLYKmu9BAKP/gUnrpupFCr6enTusfog/01E/
```
Comment (fsdiogo) [inline src/stats-view/style.js]: This is a bit odd, a `.padding` class that changes the margin 😛

**Review by hugomrdias — APPROVED** [review]

**Review by fsdiogo — APPROVED** [review]
👍

**Pull request merged**